packages/editor/src/lib/exports/getSvgJsx.tsx

```diff
@@ -0,0 +1,294 @@
+import {
+	TLFrameShape,
+	TLGroupShape,
+	TLShape,
+	TLShapeId,
+	getDefaultColorTheme,
+} from '@tldraw/tlschema'
+import { ComponentType, Fragment, ReactElement } from 'react'
+import { ErrorBoundary } from '../components/ErrorBoundary'
+import { InnerShape, InnerShapeBackground } from '../components/Shape'
+import { Editor } from '../editor/Editor'
+import { ShapeUtil } from '../editor/shapes/ShapeUtil'
+import {
+	SvgExportContext,
+	SvgExportContextProvider,
+	SvgExportDef,
+} from '../editor/types/SvgExportContext'
+import { TLSvgOptions } from '../editor/types/misc-types'
+import { useEditor } from '../hooks/useEditor'
+import { Box } from '../primitives/Box'
+import { Mat } from '../primitives/Mat'
+
+export async function getSvgJsx(editor: Editor, ids: TLShapeId[], opts: TLSvgOptions = {}) {
+	if (!window.document) throw Error('No document')
+
+	const {
+		scale = 1,
+		background = false,
+		padding = editor.options.defaultSvgPadding,
+		preserveAspectRatio = false,
+	} = opts
+
+	const isDarkMode = opts.darkMode ?? editor.user.getIsDarkMode()
+	const theme = getDefaultColorTheme({ isDarkMode })
+
+	// ---Figure out which shapes we need to include
+	const shapeIdsToInclude = editor.getShapeAndDescendantIds(ids)
+	const renderingShapes = editor
+		.getUnorderedRenderingShapes(false)
+		.filter(({ id }) => shapeIdsToInclude.has(id))
+
+	// --- Common bounding box of all shapes
+	let bbox = null
+	if (opts.bounds) {
+		bbox = opts.bounds
+	} else {
+		for (const { id } of renderingShapes) {
+			const maskedPageBounds = editor.getShapeMaskedPageBounds(id)
+			if (!maskedPageBounds) continue
+			if (bbox) {
+				bbox.union(maskedPageBounds)
+			} else {
+				bbox = maskedPageBounds.clone()
+			}
+		}
+	}
+
+	// no unmasked shapes to export
+	if (!bbox) return
+
+	const singleFrameShapeId =
+		ids.length === 1 && editor.isShapeOfType<TLFrameShape>(editor.getShape(ids[0])!, 'frame')
+			? ids[0]
+			: null
+	if (!singleFrameShapeId) {
+		// Expand by an extra 32 pixels
+		bbox.expandBy(padding)
+	}
+
+	// We want the svg image to be BIGGER THAN USUAL to account for image quality
```

Comment:
bigger than usual, because of retina? or some other reason?

Reply:
not sure - this was already in this file